mlir/utils/jenkins/Jenkinsfile

```diff
@@ -360,6 +360,24 @@ void archivePerfDB() {
     onlyIfSuccessful: true
 }
 
+boolean shouldRunBuildAndTest(String codepath) {
```

Comment:
Pull the big condition out of the when-block and into this function.  Avoids method-too-large error when I add the code-coverage stage.

Reply:
Should we pull this refactor into other spots in the Jenkinsfile?